pkg/distributor/distributor.go

```diff
@@ -816,7 +824,7 @@ func (d *Distributor) doBatch(ctx context.Context, req *cortexpb.WriteRequest, s
 			}
 		}
 
-		return d.send(localCtx, ingester, timeseries, metadata, req.Source)
+		return d.send(localCtx, ingester, timeseries, metadata, req.Source, stats)
```

Comment:
Do we need to sum the samples pulled from `stats`? Now I see we just overwrite stats for every request.

Reply:
If so, isn't there a good chance the returned header value (`X-Prometheus-Remote-Write-Samples-Written`) would be multiple of samples in a write request?

Reply:
Yeah. With replication factor it is expected to have more samples. I think this is fine.